aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START aiplatform_gemini_get_started]
+
+using Google.Api.Gax.Grpc;
+using Google.Cloud.AIPlatform.V1;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+public class GeminiQuickstart
+{
+    public async Task<string> GenerateContent(
+        string projectId = "your-project-id",
+        string location = "us-central1",
+        string publisher = "google",
+        string model = "gemini-pro-vision"
+    )
+    {
+        // Create client
+        var predictionServiceClient = new PredictionServiceClientBuilder
+        {
+            Endpoint = $"{location}-aiplatform.googleapis.com"
+        }.Build();
+
+        // Prompt
+        string prompt = "What's in this photo";
+        string imageUri = "gs://generativeai-downloads/images/scones.jpg";
+
+        // Initialize request argument(s)
+        var content = new Content
+        {
+            Role = "USER"
+        };
+        content.Parts.AddRange(new List<Part>()
+        {
+            new() {
+                Text = prompt
+            },
+            new() {
+                FileData = new() {
+                    MimeType = "image/png",
+                    FileUri = imageUri
+                }
+            }
+        });
+
+        var generateContentRequest = new GenerateContentRequest
+        {
+            Model = $"projects/{projectId}/locations/{location}/publishers/{publisher}/models/{model}",
```

Comment:
I'm guessing we don't have a resource name helper for this? Something like ModelName.FromProjectLocationPublisher...?

Reply:
Nope, no such resource helper. There should be one, maybe we can open a bug?